kernel/src/main/scala/org/apache/toree/boot/layer/ComponentInitialization.scala

```diff
@@ -138,35 +137,47 @@ trait StandardComponentInitialization extends ComponentInitialization {
     commManager: CommManager,
     pluginManager: PluginManager
   ) = {
+
+    //kernel has a dependency on ScalaInterpreter to get the ClassServerURI for the SparkConf
+    //we need to pre-start the ScalaInterpreter
+    val scalaInterpreter = interpreterManager.interpreters("Scala")
+    scalaInterpreter.start()
+
     val kernel = new Kernel(
       config,
       actorLoader,
       interpreterManager,
       commManager,
       pluginManager
-    )
-    /*
-    interpreter.doQuietly {
-      interpreter.bind(
-        "kernel", "org.apache.toree.kernel.api.Kernel",
-        kernel, List( """@transient implicit""")
-      )
+    ){
+      override protected[toree] def createSparkConf(conf: SparkConf) = {
+        val theConf = super.createSparkConf(conf)
+
+        // TODO: Move SparkIMain to private and insert in a different way
+        logger.warn("Locked to Scala interpreter with SparkIMain until decoupled!")
+
+        // TODO: Construct class server outside of SparkIMain
+        logger.warn("Unable to control initialization of REPL class server!")
+        logger.info("REPL Class Server Uri: " + scalaInterpreter.classServerURI)
+        conf.set("spark.repl.class.uri", scalaInterpreter.classServerURI)
+
+        theConf
+      }
     }
-    */
     pluginManager.dependencyManager.add(kernel)
 
     kernel
   }
 
   private def createPluginManager(
-    config: Config, interpreter: Interpreter,
+    config: Config, interpreterManager: InterpreterManager,
     dependencyDownloader: DependencyDownloader
   ) = {
     logger.debug("Constructing plugin manager")
     val pluginManager = new PluginManager()
 
     logger.debug("Building dependency map")
-    pluginManager.dependencyManager.add(interpreter)
+    pluginManager.dependencyManager.add(interpreterManager.interpreters.get("Scala").orNull)
```

Comment:
Similar to default interpreter question. Do we want to just pass null here? Or fail explicitly? Otherwise, having a missing Scala interpreter results in plugins down the line failing with a mysterious null pointer exception.